### PR TITLE
Raycaster: Option to skip invisible objects

### LINKED ISSUE #3647: Intersect should not pick invisible objects
#2277 was closed without comment.  A ray should not intersect invisible objects; a cursory reading of the source doesn't appear to show any check for visible at all in that part of the code.  And yes, I realize you could iterate the returned objects but I still maintain that it shouldn't be checking invisible objects at all.  Or at least it could be a parameter somewhere.

### PR DESCRIPTION
**Pull request by dubejf**
Fixes #3647

### CODE REVIEW

**Comment by dubejf** [comment]
Oops! Also include fix for #3385.

File: docs/api/core/Raycaster.html [hunk 1]
```diff
@@ -98,6 +98,11 @@
 		This value shouldn't be negative and should be larger than the near property. 
 		</div>
 
+		<h3>[property:boolean visibleOnly]</h3>
+		<div>
+		Defines whether the raycaster should intersect only visible objects. Default is false.
```
Comment (WestLangley) [inline docs/api/core/Raycaster.html]: I know it is not backwards-compatible, but defaulting `visibleOnly` to `true` seems more natural to me. What do you think?

Comment (dubejf) [inline docs/api/core/Raycaster.html]: Done.

**Comment by mrdoob** [comment]
The additional changes look good, but I don't think I like the `visibleOnly` feature.

**Comment by WestLangley** [comment]
I do not have a strong feeling one way or the other. Prior to this PR, Raycaster detects invisible objects, too.

BTW, we also have the `material.visible` property. Whatever we do, we should be consistent across the board.

**Comment by mrdoob** [comment]
@dubejf could you do a new PR with the changes not related to `visibleOnly`?

**Pull request closed without merging**